core/src/main/java/org/apache/james/core/Domain.java

```diff
@@ -39,7 +47,7 @@ public static Domain of(String domain) {
 
     protected Domain(String domainName) {
         this.domainName = domainName;
-        this.normalizedDomainName = domainName.toLowerCase(Locale.US);
+        this.normalizedDomainName = removeBrackets(domainName.toLowerCase(Locale.US));
```

Comment:
here, you assume it's domain responsibility to remove angle brackets from the input string. I really think this "quoting" is related to emails and should not be Domain responsibility.